crates/wasmtime/src/instance.rs

```diff
@@ -591,18 +637,16 @@ impl<'a> Instantiator<'a> {
                 // means we need to handle that here, otherwise we defer to the
                 // instance to load the values.
                 let item = match index {
-                    EntityIndex::Instance(i) => {
-                        wasmtime_runtime::Export::Instance(self.cur.instances[*i].clone())
-                    }
-                    EntityIndex::Module(i) => {
-                        wasmtime_runtime::Export::Module(Box::new(self.cur.modules[*i].clone()))
-                    }
-                    index => instance.handle.lookup_by_declaration(index),
+                    EntityIndex::Instance(i) => Extern::Instance(self.cur.instances[*i].clone()),
+                    EntityIndex::Module(i) => Extern::Module(self.cur.modules[*i].clone()),
+                    index => unsafe {
+                        Extern::from_wasmtime_export(instance.lookup_by_declaration(index), store)
+                    },
                 };
                 (name.clone(), item)
             })
             .collect();
```

Comment:
R.e. `Instance::exports` above, this loop and this `collect()` are what I'd like to avoid in the future. This is an unnecessary allocation that instantiation performs which I think we can get rid of at no extra cost.